### PR TITLE
[muxorch][202205] Fixing cache bug in updateRoute logic

### PR DESCRIPTION
**Pull request by Ndancejic**
**What I did**
Fixed a bug where cache was not removing entry when an ecmp mux route gets updated to a single nexthop route. Removed cache from updateRoute logic, and relying on other logic to skip over multiple mux nexthop case.

**Why I did it**
Mux switchover was failing because it was skipping over the route in updateNextHopRoutes.

**How I verified it**
Added vstest and ran test cases on virtual switch

**Details if related**
cherry-pick of https://github.com/sonic-net/sonic-swss/pull/2982

MSFT ADO - 26099588

### CODE REVIEW

File: orchagent/routeorch.cpp [hunk 1]
```diff
@@ -2185,7 +2182,24 @@ bool RouteOrch::addRoutePost(const RouteBulkContext& ctx, const NextHopGroupKey
             if (ol_nextHops.getSize() > 1
                 && m_syncdNextHopGroups[ol_nextHops].ref_count == 0)
             {
-                m_bulkNhgReducedRefCnt.emplace(ol_nextHops, 0);
+                if (m_syncdNextHopGroups[ol_nextHops].ref_count == 0)
```
Comment (prsunny) [inline orchagent/routeorch.cpp:2185]: This seems to be unrelated to your PR. Can you please revert?

Comment (Ndancejic) [reply orchagent/routeorch.cpp:2185]: Oh yes, this is what I was talking about earlier. I forgot to commit, let me fix that

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
Failure on test-chassis, unrelated to PR changes. Merging to unblock build
